src/components/reservation/reservation-modal.tsx

```diff
@@ -94,11 +98,23 @@ const ReservationModal = ({ manifestation }: ReservationModalProps) => {
       }
     };
     mutate(batch, {
-      onSuccess: () => setReservationDidSuccess(true),
-      onError: () => setReservationDidSuccess(false)
+      onSuccess: (res) => {
+        // this state is used to show the success or error modal
+        setReservationResponse(res);
+        // because after a successful reservation the holdings (reservations) are updated
+        queryClient.invalidateQueries(getGetHoldingsV3QueryKey());
+      }
     });
   };
 
+  const reservationResult = reservationResponse?.reservationResults
+    ? reservationResponse.reservationResults[0].result
+    : null;
```

Comment:
Please get the result of the reservation operation as a boolean from `reservationResponse.success`.

This is more clean.